src/chrome/crdpMultiplexing/crdpMultiplexor.ts

```diff
@@ -0,0 +1,201 @@
+        if (this._enabledDomains[domain]) {
+            this.callMessageCallbacks(messageData);
+        } else if (this._pendingMessagesForDomain !== null) {
+            // We give clients 60 seconds after they connect to the channel to enable domains and receive all messages
+            this.storeMessageForLater(domain, messageData);
+        }
+    }
+
+    private storeMessageForLater(domain: string, messageData: string): void {
+        let messagesForDomain = this._pendingMessagesForDomain[domain];
+        if (messagesForDomain === undefined) {
+            this._pendingMessagesForDomain[domain] = [];
+            messagesForDomain = this._pendingMessagesForDomain[domain];
+        }
+
+        // Usually this is too much logging, but we might use it while debugging
+        // logger.log(`CRDP Multiplexor - Storing message to channel ${this.name} for ${domain} for later: ${messageData}`);
+        messagesForDomain.push(messageData);
+    }
+
+    constructor(public name: string, public id: number, private _multiplexor: CRDPMultiplexor) { }
+
+    public send(messageData: string): void {
+        const message = JSON.parse(messageData);
+        const method = message.method;
+        const isEnableMethod = method && method.indexOf(".enable") >= 0;
+        let domain;
+
+        if (isEnableMethod) {
+            domain = extractDomain(method);
+            this._enabledDomains[domain] = true;
+        }
+
+        this._multiplexor.send(this, messageData);
+
+        if (isEnableMethod) {
+            this.sendUnsentPendingMessages(domain);
+        }
+    }
+
+    private sendUnsentPendingMessages(domain: string): void {
+        const pendingMessagesData = this._pendingMessagesForDomain[domain];
+        if (pendingMessagesData !== undefined && this._messageCallbacks.length) {
+            logger.log(`CRDP Multiplexor - Sending pending messages of domain ${domain}(Count = ${pendingMessagesData.length})`);
+            delete this._pendingMessagesForDomain[domain];
+            pendingMessagesData.forEach(pendingMessageData => {
+                this.callDomainMessageCallbacks(domain, pendingMessageData);
+            });
+        }
+    }
+
+    private discardUnsentPendingMessages(): void {
+        logger.log(`CRDP Multiplexor - Discarding unsent pending messages for domains: ${Object.keys(this._pendingMessagesForDomain).join(", ")}`);
+        this._pendingMessagesForDomain = null;
+    }
+
+    public on(event: string, cb: Function): void;
+    public on(event: "open", cb: (ws: LikeSocket) => void): void;
+    public on(event: "message", cb: (data: string) => void): void;
+    public on(event: string, cb: Function): void {
+        if (event === 'message') {
+            if (this._messageCallbacks.length === 0) {
+                setTimeout(() => this.discardUnsentPendingMessages(), CRDPChannel.timeToPreserveMessagesInMillis);
+            }
+
+            this._messageCallbacks.push(cb);
```

Comment:
I guess you could, and here you can have more than one connection on the websocket server.